docs/concepts/linq.rst

```diff
@@ -0,0 +1,268 @@
+	                   where !prop1Value.Equals(prop2Value)
+	                   select new { Prop =  prop1.Name, prop1Value, prop2Value }
+	
+	if (unequalProps.Any())
+	{
+	    // The two objects differ!
+	    foreach (var unequalSet in unequalProps)
+	    {
+	        Console.WriteLine($"Unequal Property {unequalSet.Prop} - {prop1Value} != {prop2value}");
+	    }
+	}
+				  
+Why Use the Query Syntax?
+-------------------------
+
+This is a question which often comes up.  After all, this,
+
+.. code-block:: c#
+
+	var filteredItems = myItems.Where(item => item.Foo);
+	
+is a lot more concise than this:
+
+.. code-block:: c#
+
+	var filteredItems = from item in myItems
+	                    where item.Foo
+	                    select item;
+
+Isn't the API syntax just a more concise way to do the query syntax?
+
+**Short answer:** 
+
+No, and the query syntax is more powerful.
+
+**Long answer:**
+
+The query syntax allows for the use the **let** clause, which allows you to introduce a variable within the scope of the expression, using it in subsequent pieces of the expression.  Such behavior can be annoying to reproduce using the API syntax.  This is best demonstrated via example:
+
+TODO: More goes here
+
+So this begs the question, **should you just use the query syntax?**
+
+The answer to this question is **yes** if...
+
+* Your existing codebase already uses the query syntax
+* You need to scope variables within your queries due to complexity
+* You prefer the query syntax and it won't distract from your codebase
+
+The answer to this question is **no** if...
+
+* Your existing codebase already uses the API syntax
+* You have no need to scope variables within your queries
+* You prefer the API syntax and it won't distract from your codebase
+
+PLINQ
+-----
+
+PLINQ, or Parallel LINQ, is a parallel execution engine for LINQ expressions.  In other words, a regular LINQ expressions can be trivially parallelized across any number of threads.  This is accomplished via a call to ``AsParallel()`` preceding the expression.
+
+Consider the following:
+
+.. code-block:: c#
+
+	public static string GetAllFacebookUserLikes(facebookUsers)
+	{
+	    var seed = default(UInt64);
+		
+	    Func<UInt64, UInt64, UInt64> threadAccumulator = (t1, t2) => t1 + t2;
+	    Func<UInt64, UInt64, UInt64> threadResultAccumulator = (t1, t2) => t1 + t2;
+	    Func<Uint64, string> resultSelector = total => $"Facebook has {total} likes!";
+	    
+	    return facebookUsers.AsParallel()
+	                        .Aggregate(seed, threadAccumulator, threadResultAccumulator, resultSelector);
+	}
+									   
+This code will partition ``facebookUsers`` across system threads as necessary, sum up the total likes on each thread in parallel, sum the results **of** each thread, and project that result into a nice string.
+
+Or in diagram form:
+
+.. image:: primer/_static/PLINQ-diagram.png
+	:align: center
+	
+Further Resources:
+------------------
+
+THEY GO HERE
```

Comment:
Link to LINQ pad would be good here, 101 samples, the http://blogs.msdn.com/b/mattwar/archive/2007/07/30/linq-building-an-iqueryable-provider-part-i.aspx series also.